subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/adapter/ProtocolToModelAdapter.java

```diff
@@ -324,7 +325,7 @@ public Object unpack(Object viewObject) {
 
     private static class ViewGraphDetails implements Serializable {
         // Transient, don't serialize all the views that happen to have been visited, recreate them when visited via the deserialized view
-        private transient Map<ViewKey, Object> views = new HashMap<ViewKey, Object>();
+        private transient Map<ViewKey, Object> views = new WeakHashMap<>();
```

Comment:
🤔 At first glance, `WeakHashMap<ViewKey, ...>` doesn't seem to capture the semantics of this cache well. `ViewKey` instances are short-lived by design.

Reply:
Under `ViewKey instances are short-lived by design` you mean that we don't keeping references at `ViewKey`'s? 
This is a true, but being a part of `HashMap` is preventing entries to be garbage collected.

Or you mean that `ViewGraphDetails` should be collected entirely?

Reply:
> Under `ViewKey instances are short-lived by design` you mean that we don't keeping references at `ViewKey`'s?
> This is a true, but being a part of `HashMap` is preventing entries to be garbage collected.

Yes, AFAICS `ViewKey` instances are not retained so there's no guarantee any values would be cached at all, or worse, there are no guarantees identities would be preserved across the mapping (which I'm not sure matters in this particular case but needs to be considered).